specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/preview/2019-02-01-preview/imagebuilder.json

```diff
@@ -0,0 +1,1148 @@
+          "description": "URI to get the ISO image. This URI has to be accessible to the resource provider at the time of the image template creation."
+        },
+        "sha256Checksum": {
+          "type": "string",
+          "description": "SHA256 Checksum of the ISO image."
+        }
+      },
+      "required": [
+        "sourceURI",
+        "sha256Checksum"
+      ]
+    },
+    "ImageTemplatePlatformImageSource": {
+      "description": "Describes an image source from [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages).",
+      "x-ms-discriminator-value": "PlatformImage",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ImageTemplateSource"
+        }
+      ],
+      "properties": {
+        "publisher": {
+          "type": "string",
+          "description": "Image Publisher in [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        },
+        "offer": {
+          "type": "string",
+          "description": "Image offer from the [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        },
+        "sku": {
+          "type": "string",
+          "description": "Image sku from the [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        },
+        "version": {
+          "type": "string",
+          "description": "Image version from the [Azure Gallery Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages)."
+        }
+      }
+    },
+    "ImageTemplateManagedImageSource": {
+      "description": "Describes an image source that is a managed image in customer subscription.",
+      "x-ms-discriminator-value": "ManagedImage",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ImageTemplateSource"
+        } 
+      ],
+      "properties": {
+        "imageId": {
+          "type": "string",
+          "description": "ARM resource id of the managed image in customer subscription"
+        } 
+      },
+      "required": [
+        "imageId"
+      ]
+    },
+    "ImageTemplateCustomizer": {
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The type of customization tool you want to use on the Image. For example, \"Shell\" can be shell customizer"
+        },
+        "name": {
+          "type": "string",
+          "description": "Friendly Name to provide context on what this customization step does"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "ImageTemplateShellCustomizer": {
+      "description": "Runs a shell script during the customization phase (Linux)",
+      "x-ms-discriminator-value": "Shell",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ImageTemplateCustomizer"
+        }
+      ],
+      "properties": {
+        "script": {
```

Comment:
`scriptUri` or even `fileUri` might be a more descriptive name. The name alone doesn't tell me if I should supply a json encoded script contents or a URL

Reply:
we have a change coming up to allow inline scripts for `Shell` and `PowerShell`. We'll rename this property with that change.
/CC @sumit-kalra: I've added this note to the work item